Sources/CocoaLumberjack/include/CocoaLumberjack/DDLog.h

```diff
@@ -845,7 +849,10 @@ typedef NS_OPTIONS(NSInteger, DDLogMessageOptions){
 @property (readonly, nonatomic) NSString *fileName;
 @property (readonly, nonatomic, nullable) NSString * function;
 @property (readonly, nonatomic) NSUInteger line;
-@property (readonly, nonatomic, nullable) id tag;
+#if DD_LEGACY_MESSAGE_TAG
+@property (readonly, nonatomic, nullable) id tag __attribute__((deprecated("Use representedObject instead", "representedObject")));
+#endif
+@property (readonly, nonatomic, nullable) id representedObject;
```

Comment:
This line needs `#else` macro.

Reply:
The idea here is that `tag` property is from now on deprecated. And going forward the `representedObject` property is replacing it. (Not an if/else case)

In case somebody has a type conflict with the deprecated `tag` property, it is possible to set the `DD_LEGACY_MESSAGE_TAG` to **0**, removing the `tag` property. This is more often the case when adopting CocoaLumberjack into an existing project, not risking the chance of breaking existing CocoaLumberjack code depending on the `tag` property.

Speaking of it. It may makes sense to declare the 
```
@interface DDLogMessage ()
@property (readwrite, nonatomic, nullable) id tag __attribute__((deprecated("Use representedObject instead", "representedObject"))); // ensure compatibility even when build with DD_LEGACY_MESSAGE_TAG to 0.
@end
```
in _DDLog.m_ to ensure compatibility even when build with `DD_LEGACY_MESSAGE_TAG` to `0`. Or alternatively add it as such
```
- (id)tag {
    return _tag;
}
```

Reply:
Thank you for explaining your idea in detail. I clearly understand:)

> @interface DDLogMessage ()
> @property (readwrite, nonatomic, nullable) id tag __attribute__((deprecated("Use representedObject instead", "representedObject"))); // ensure compatibility even when build with DD_LEGACY_MESSAGE_TAG to 0.
> @end

I prefer the above.